src/test/java/plugins/GradlePluginTest.java

```diff
@@ -135,93 +192,274 @@ public void run_gradle_script_build_scan_link() {
+    public void run_gradle_user_home_workspace(){
+        GradleInstallation.installGradle(jenkins, GradleInstallation.VERSION_4_0_NAME, GradleInstallation.LATEST_VERSION);
         final FreeStyleJob job = jenkins.jobs.create();
-        job.copyResource(resource("/gradle_plugin/script.gradle"), "build.gradle");
+        job.copyResource(resource(GRADLE_SCRIPT), "build.gradle");
 
-        final String location = "test";
-        GradleWrapper.downloadWrapperFiles(job);
-        GradleWrapper.moveWrapperFiles(job, location);
-        GradleWrapper.addWrapperStep(job, location, "hello");
+        final GradleStep step = job.addBuildStep(GradleStep.class);
+        step.setTasks(HELLO.getName());
+        step.setVersion(GradleInstallation.VERSION_4_0_NAME);
+        step.setForceGradleHomeToUseWorkspace();
+        job.save();
 
+        job.startBuild().shouldSucceed();
+        assertThat(job, Workspace.workspaceContains("caches"));
+    }
+
+    /**
+     * Verify the execution of gradle build script with committed gradle project properties.
+     */
+    @Test @WithPlugins("gradle@1.27-SNAPSHOT")
+    public void run_gradle_add_project_properties(){
+        GradleInstallation.installGradleVersion_4_0(jenkins);
+
+        final FreeStyleJob job = jenkins.jobs.create();
+        job.copyResource(resource(GRADLE_SCRIPT), "build.gradle");
+
+        final GradleStep step = job.addBuildStep(GradleStep.class);
+        step.setVersion(GradleInstallation.VERSION_4_0_NAME);
+        step.setProjectProperties("TEST_PARAM_1=hello\nTEST_PARAM_2=world");
+        step.setTasks(JOB_PARAM_AS_PROJECT_PROPERTIES.getName());
         job.save();
+
         final Build build = job.startBuild().shouldSucceed();
+        assertThat(build.getConsole(), containsString("Gradle Properties: hello world"));
+    }
 
-        assertThat(build.getConsole(), containsString(SUCCESSFUL_BUILD));
-        assertThat(build.getConsole(), containsString("Hello world!"));
+    /**
+     * Verify the execution of gradle build script with committed gradle system properties.
+     */
+    @Test @WithPlugins("gradle@1.27-SNAPSHOT")
+    public void run_gradle_add_system_properties(){
+        GradleInstallation.installGradleVersion_4_0(jenkins);
+        final FreeStyleJob job = jenkins.jobs.create();
+        job.copyResource(resource(GRADLE_SCRIPT), "build.gradle");
+
+        final GradleStep step = job.addBuildStep(GradleStep.class);
+        step.setVersion(GradleInstallation.VERSION_4_0_NAME);
+        step.setSystemProperties("TEST_PARAM_1=hello\nTEST_PARAM_2=world");
+        step.setTasks(JOB_PARAM_AS_SYSTEM_PROPERTIES.getName());
+        job.save();
+
+        final Build build = job.startBuild().shouldSucceed();
+        assertThat(build.getConsole(), containsString("Gradle Properties: hello world"));
+    }
+
+    /**
+     * Verify the existence of links for executed tasks.
+     */
+    @Test @WithPlugins("gradle@1.27-SNAPSHOT")
```

Comment:
The Versioning was somewhat broken but should work now.
The Annotations should not be needed anymore.
I thought, they did not do any harm so i left them in. I can remove them if you want.